cylc/flow/network/ssh_client.py

```diff
@@ -0,0 +1,103 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Union
+
+from cylc.flow.suite_files import load_contact_file, ContactFileFields
+from cylc.flow.network import (
+    get_location,
+)
+import json
+from cylc.flow.remote import _remote_cylc_cmd
+from cylc.flow.exceptions import ClientError
+
+
+class SuiteRuntimeClient():
+    """Client to the workflow server communication using ssh.
+
+    Determines host from the contact file unless provided.
+
+    Args:
+        suite (str):
+            Name of the suite to connect to.
+        timeout (float):
+            Set the default timeout in seconds.
+            See: https://github.com/cylc/cylc-flow/issues/4112
+        host (str):
+            The host where the flow is running if known.
+
+            If host is provided it is not necessary to load
+            the contact file.
+    """
+    def __init__(
+            self,
+            suite: str,
+            timeout: Union[float, str] = None,
+            host: str = None,
```

Comment:
Could we swap these around so that `suite` and `host` use the same order as `cylc.flow.network.client.SuiteRuntimeClient`:

```suggestion
            host: str = None,
            timeout: Union[float, str] = None
```